htlcswitch/link.go

```diff
@@ -2481,11 +2484,31 @@ func (l *channelLink) sendMalformedHTLCError(htlcIndex uint64,
 }
 
 // fail helper function which is used to encapsulate the action necessary for
-// proper disconnect.
-func (l *channelLink) fail(format string, a ...interface{}) {
+// proper disconnect. If forceClose is true, we will in addition to
+// disconnecting the peer unilateraly close the channel.
+func (l *channelLink) fail(forceClose bool, format string, a ...interface{}) {
+	// TODO(halseth): optionally send error to remote?
 	reason := errors.Errorf(format, a...)
-	log.Error(reason)
+	l.errorf("Failing link: %s", reason)
+
+	l.cfg.Peer.WipeChannel(l.channel.ChannelPoint())
```

Comment:
IIRC, the `Disconnect` method should wipe the channel itself? Or even once the peer disconnects, then the `peerTerminationWatcher` will remove it from the link all together.

Reply:
Yep, removed it.